server/src/main/java/org/elasticsearch/repositories/SnapshotShardContext.java

```diff
@@ -79,6 +80,12 @@ public SnapshotShardContext(
         this.snapshotStartTime = snapshotStartTime;
     }
 
+    private static SubscribableListener<ShardSnapshotResult> createListener(ActionListener<ShardSnapshotResult> listener) {
+        final SubscribableListener<ShardSnapshotResult> objectSubscribableListener = new SubscribableListener<>();
```

Comment:
why the name `objectSubscribableListener`? Indeed could we not do `super(new SubscribableListener<>())` and then `addListener(commitRef.closingBefore(listener))` in the constructor?